persistence/module.go

```diff
@@ -204,19 +205,18 @@ func initializeBlockStore(blockStorePath string) (kvstore.KVStore, error) {
 	return kvstore.NewKVStore(blockStorePath)
 }
 
-// TODO(drewsky): Simplify and externalize the logic for whether genesis should be populated and
-// move the if logic out of this file.
+// HACK(olshansky): Simplify and externalize the logic for whether genesis should be populated and
+//                  move the if logic out of this file.
 func (m *persistenceModule) shouldHydrateGenesisDb() (bool, error) {
 	checkContext, err := m.NewReadContext(-1)
 	if err != nil {
 		return false, err
 	}
 	defer checkContext.Close()
 
-	maxHeight, err := checkContext.GetLatestBlockHeight()
-	if err == nil || maxHeight == 0 {
+	_, err = checkContext.GetLatestBlockHeight()
+	if err != nil {
```

Comment:
```suggestion
```
Only if suggestion above is accepted

Reply:
In the future, note that you can do a "multiline suggestion" by highlighting and dragging down multiple lines.

![record](https://user-images.githubusercontent.com/1892194/200444318-ad3e2244-a5c0-4038-9df5-101b4026cbe7.gif)